### DISCUSSION TITLE
Why does react-docgen-typescript not generate `args` types for external modules?

### DISCUSSION THREAD

**Discussion by suparngp** [storybookjs/storybook#12600] [Help] [answered]
I have a monorepo setup with typescript with two packages `components` and `stories`. The `stories` use `components` as a dependency and export all the stories. The `components` package is a shared component library which gets used by other packages as well. Both the packages are compiled as `ESNext` packages. When I use a component in story, I can import all the types. So I am assuming that typescript compiler did the right thing.

However, when I use `useArgs` hoping that the Args types will be automatically generated, they aren't. In fact there are no args at all by default. Can someone confirm if this is supposed to even work or not?

**Comment by shilman** [comment, marked as answer]
This doesn't currently work https://github.com/storybookjs/storybook/issues/12185

As for why, I don't think anybody's looked into it yet. cc @hipstersmoothie

**Reply by Shusanta** [reply]
@shilman any updates?
